hip_data_tools/etl/google_sheet_to_athena.py

```diff
@@ -169,17 +190,43 @@ def load_sheet_to_athena(self, overwrite_table=False):
         sheet_util = self._get_sheets_util()
         athena_util = self._get_athena_util()
         s3_util = self._get_s3_util()
+
         if overwrite_table:
             athena_util.drop_table(self.settings.table_name)
             s3_util.delete_recursive(self.settings.s3_dir)
+
+        field_names = []
+        field_types = []
+        if self.settings.fields is None:
+            field_names = sheet_util.get_fields_names(self.settings.workbook_name,
+                                                      self.settings.sheet_name)
+            field_types = sheet_util.get_fields_types(self.settings.workbook_name,
+                                                      self.settings.sheet_name)
+            self.__validate_field_names(field_names)
+        else:
+            for field in self.settings.fields:
+                field_name_type = field.split(':')
+                field_name = field_name_type[0]
+                field_names.append(field_name)
+                field_types.append(field_name_type[1])
+                self.__validate_field_names([field_name])
+
         values_matrix = sheet_util \
             .get_value_matrix(workbook_name=self.settings.workbook_name,
                               sheet_name=self.settings.sheet_name,
                               row_range=self.settings.row_range,
                               skip_top_rows_count=self.settings.skip_top_rows_count)
         log.info("The value matrix:\n %s", values_matrix)
-        table_settings = self._get_table_settings()
+        table_settings = self._get_table_settings(field_names, field_types)
         athena_util.create_table(table_settings)
-        insert_query = self._get_the_insert_query(values_matrix=values_matrix)
+        insert_query = self._get_the_insert_query(values_matrix=values_matrix, types=field_types)
         log.info("The insert query:\n %s", insert_query)
         athena_util.run_query(query_string=insert_query)
+
+    @staticmethod
+    def __validate_field_names(field_names):
+        for field_name in field_names:
+            # check for strings which only contains letters, numbers and underscores
+            if not re.match("^[A-Za-z0-9_]*$", field_name):
```

Comment:
are empty field names accepted? otherwise maybe switch to the below
```suggestion
            if not re.match("^[A-Za-z0-9_]+$", field_name):
```

Reply:
also do we need to check that there is no doubling up on the same name?